sparta.go

```diff
@@ -933,6 +943,20 @@ func NewLambda(roleNameOrIAMRoleDefinition interface{},
 	return lambda
 }
 
+// NewNamedLambda returns a LambdaAWSInfo value with a declared function name that can be provisioned
+// via CloudFormation. The `functionName` value must be a non-empty string, otherwise CloudFormation
+// will error out as "" is not a valid function name. The roleNameOrIAMRoleDefinition must either be
+// a `string` or `IAMRoleDefinition` type
+func NewNamedLambda(roleNameOrIAMRoleDefinition interface{},
+fn LambdaFunction,
+lambdaOptions *LambdaFunctionOptions,
+functionName string) *LambdaAWSInfo {
```

Comment:
I have updated the code to check for an empty functionName parameter in NewNamedLambda. The function now returns an additional error value indicating if an empty functionName was passed and nil otherwise.